### PR TITLE
updated example of AST for TokenTypes.QUESTION

### PR DESCRIPTION
**Pull request by RuchiGitHu**
Closes : #9557 
image

$ javac test.java

$ cat test.java
public class test {
public static void main() {
int quantity=0;
String variable=(quantity==1)?"true":"false";
}
}

$ java -jar checkstyle-8.41-all.jar -T test.java
CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|   `--LITERAL_PUBLIC -> public [1:0]
|--LITERAL_CLASS -> class [1:7]
|--IDENT -> test [1:13]
`--OBJBLOCK -> OBJBLOCK [1:17]
|--LCURLY -> { [1:17]
    |--METHOD_DEF -> METHOD_DEF [2:0]
    |   |--MODIFIERS -> MODIFIERS [2:0]
    |   |   |--LITERAL_PUBLIC -> public [2:0]
    |   |   `--LITERAL_STATIC -> static [2:7]
    |   |--TYPE -> TYPE [2:14]
    |   |   `--LITERAL_VOID -> void [2:14]
    |   |--IDENT -> main [2:19]
    |   |--LPAREN -> ( [2:23]
    |   |--PARAMETERS -> PARAMETERS [2:24]
    |   |--RPAREN -> ) [2:24]
    |   `--SLIST -> { [2:25]
    |       |--VARIABLE_DEF -> VARIABLE_DEF [3:0]
    |       |   |--MODIFIERS -> MODIFIERS [3:0]
    |       |   |--TYPE -> TYPE [3:0]
    |       |   |   `--LITERAL_INT -> int [3:0]
    |       |   |--IDENT -> quantity [3:4]
    |       |   `--ASSIGN -> = [3:12]
    |       |       `--EXPR -> EXPR [3:13]
    |       |           `--NUM_INT -> 0 [3:13]
    |       |--SEMI -> ; [3:14]
    |       |--VARIABLE_DEF -> VARIABLE_DEF [4:0]
    |       |   |--MODIFIERS -> MODIFIERS [4:0]
    |       |   |--TYPE -> TYPE [4:0]
    |       |   |   `--IDENT -> String [4:0]
    |       |   |--IDENT -> variable [4:7]
    |       |   `--ASSIGN -> = [4:15]
    |       |       `--EXPR -> EXPR [4:29]
    |       |           `--QUESTION -> ? [4:29]
    |       |               |--LPAREN -> ( [4:16]
    |       |               |--EQUAL -> == [4:25]
    |       |               |   |--IDENT -> quantity [4:17]
    |       |               |   `--NUM_INT -> 1 [4:27]
    |       |               |--RPAREN -> ) [4:28]
    |       |               |--STRING_LITERAL -> "true" [4:30]
    |       |               |--COLON -> : [4:36]
    |       |               `--STRING_LITERAL -> "false" [4:37]
    |       |--SEMI -> ; [4:44]
    |       `--RCURLY -> } [5:0]
    `--RCURLY -> } [6:0]

Printing a line for the code we care
$ java -jar checkstyle-8.41-all.jar -T test.java | grep "4:"
 |--VARIABLE_DEF -> VARIABLE_DEF [4:0]
    |       |   |--MODIFIERS -> MODIFIERS [4:0]
    |       |   |--TYPE -> TYPE [4:0]
    |       |   |   `--IDENT -> String [4:0]
    |       |   |--IDENT -> variable [4:7]
    |       |   `--ASSIGN -> = [4:15]
    |       |       `--EXPR -> EXPR [4:29]
    |       |           `--QUESTION -> ? [4:29]
    |       |               |--LPAREN -> ( [4:16]
    |       |               |--EQUAL -> == [4:25]
    |       |               |   |--IDENT -> quantity [4:17]
    |       |               |   `--NUM_INT -> 1 [4:27]
    |       |               |--RPAREN -> ) [4:28]
    |       |               |--STRING_LITERAL -> "true" [4:30]
    |       |               |--COLON -> : [4:36]
    |       |               `--STRING_LITERAL -> "false" [4:37]
    |       |--SEMI -> ; [4:44]

Expected update for javadoc

|--VARIABLE_DEF -> VARIABLE_DEF
       |   |--MODIFIERS -> MODIFIERS
       |   |--TYPE -> TYPE
       |   |   `--IDENT -> String
       |   |--IDENT -> variable
       |   `--ASSIGN -> =
       |       `--EXPR -> EXPR
       |           `--QUESTION -> ?
       |               |--LPAREN -> (
       |               |--EQUAL -> ==
       |               |   |--IDENT -> quantity
       |               |   `--NUM_INT -> 1
       |               |--RPAREN -> )
       |               |--STRING_LITERAL -> "true"
       |               |--COLON -> :
       |               `--STRING_LITERAL -> "false"
       |--SEMI -> ;

### CODE REVIEW

**Comment by RuchiGitHu** [comment]
@romani Some checks are failed, please help me on this and let me know if there is any changes required...

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -2602,24 +2602,26 @@
...
+     *     `--QUESTION -&gt; ?
+     *  |--LPAREN -&gt; (
+     *  |--EQUAL -&gt; ==
+     *  |   |--IDENT -&gt; quantity
+     *  |   `--NUM_INT -&gt; 1
+     *  |--RPAREN -&gt; )
+     *  |--STRING_LITERAL -&gt; "true"
+     *  |--COLON -&gt; :
+     *      `--STRING_LITERAL -&gt; "false"
+     *  --SEMI -&gt; ;
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:2623]: Please copy ast structure from CLI output without modifications, please read good practice link and review commits for similar task that are already merged for a lot of examples

**Review by romani — CHANGES_REQUESTED** [review]
Item:

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (RuchiGitHu) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:2623]: Thanks for your time @romani for reviewing my PR, I will correct it as required, apologize for inconvenience caused by me.

Comment (RuchiGitHu) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: > Please copy ast structure from CLI output without modifications, please read good practice link and review commits for similar task that are already merged for a lot of examples

Please review @romani, I corrected ast structure and also PR according to that.
Thanks in advance..

Comment (RuchiGitHu) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: > > Please copy ast structure from CLI output without modifications, please read good practice link and review commits for similar task that are already merged for a lot of examples
> 
> Please review @romani, I corrected ast structure and also PR according to that.
> Thanks in advance..

Please review @romani...
Thanks in advance..

**Comment by romani** [comment]
Squash all commits in one

**Review by romani — APPROVED** [review]

**Review by romani — CHANGES_REQUESTED** [review]
content is good
but you have to sqaush commits in one.
to pass CI

**Comment by RuchiGitHu** [comment]
> content is good
> but you have to sqaush commits in one.
> to pass CI

Yes of course @romani. I'll do it as soon as possible....

**Comment by RuchiGitHu** [comment]
> content is good
> but you have to sqaush commits in one.
> to pass CI

Please review @romani, I have squash the commits.

**Comment by romani** [comment]
and now fix name of commit to let `mvn verify` to pass.

**Comment by RuchiGitHu** [comment]
> fix name of commit to let `mvn verify` to pass.

Please help me. How to achieve this?
Thanks in advance...

**Comment by romani** [comment]
please google this, there numerous post in web on how to do this by tool that you like.
`git commit --amend`

**Pull request merged**